db/migrate/20240909183413_create_categories.rb

```diff
@@ -0,0 +1,44 @@
+class CreateCategories < ActiveRecord::Migration[7.1]
+  def change
+    create_table :detectors do |t|
+      t.string :name
+      t.float :confidence
+
+      t.timestamps
+    end
+    add_index :detectors, :name, unique: true
+
+    create_table :categories do |t|
```

Comment:
Non-blocking: There's arguably a benefit to having each of these table creations be in its own migration file to allow for slightly easier context.

Reply:
I'm happy to do whatever is most expected in Rails. The migrations were originally separate files, which I combined thinking that this was one feature and set of related tables. But, that did introduce some uncertainty about the name of the migration (it is inaccurately named at the moment).